### PR TITLE
estatisticas em PDF

### PR DESCRIPTION
**Pull request by marcelobianchi**

### CODE REVIEW

File: pypqlx.py [hunk 1]
```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
...
+            minimun.append(a)
+        return
     
     def max(self):
-        pass
+        maximun =[]
+        for i in sorted(set(pdf.period)):
+            a = max(pdf.power[pdf.period == i])
+            maximun.append(a)
+        return
```
Comment (marcelobianchi) [inline pypqlx.py]: You should return a np.array(maximun)

Comment (marcelobianchi) [inline pypqlx.py]: You must return something, or better, a numpy array of the maximun values.

File: pypqlx.py [hunk 2]
```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
...
+            avg.append(a)
+        return
     
     def min(self):
-        pass
+        minimun = []
+        for i in sorted(set(pdf.period)):
+            a = min(pdf.power[pdf.period == i])
+            minimun.append(a)
+        return
```
Comment (marcelobianchi) [inline pypqlx.py]: You should return something, like the other, np.array(minimun)

File: pypqlx.py [hunk 3]
```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
...
     def median(self):
         pass
     
     def average(self):
-        pass
+        avg = []
+        for i in sorted(set(pdf.period)):
+            a = sum(pdf.power[pdf.period == i] * pdf.count[pdf.period == i]) / sum(pdf.count[pdf.period == i])
+            avg.append(a)
+        return
```
Comment (marcelobianchi) [inline pypqlx.py]: and here also, a np.array(avg)

File: pypqlx.py [hunk 4]
```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
...
+            a = pdf.power[max(pdf.count[pdf.period == i])]
+            mode.append(a)
+        return
+        
     def median(self):
         pass
     
     def average(self):
-        pass
+        avg = []
```
Comment (marcelobianchi) [inline pypqlx.py]: all the other methods you used a full variable name, do the same here.

File: pypqlx.py [hunk 5]
```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
         self.L = L if L != "" else "--"
     
     def mode(self):
-        pass
-    
+        mode = []
+        for i in sorted(set(pdf.period)):
+            a = pdf.power[max(pdf.count[pdf.period == i])]
```
Comment (marcelobianchi) [inline pypqlx.py]: You should have used the argmax doesn't ? I.e. you need to find the power where the histogram is larger for each period. Histogram amplitude is count. So, something like

```
pdf.power[pdf.period == i][np.argmax(pdf.count[pdf.period == i])]
```
 Needs to check!

File: pypqlx.py [hunk 6]
```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
         self.L = L if L != "" else "--"
     
     def mode(self):
-        pass
-    
+        mode = []
+        for i in sorted(set(pdf.period)):
```
Comment (marcelobianchi) [inline pypqlx.py]: In all changes you use pdf.SOMETHING, replace to self.SOMETHING. The pdf is a global variable and not, am internal class variable.

File: pypqlx.py (hunk 1 above)
Comment (ebrasilio) [reply pypqlx.py]: Sorry, I fixed it.

File: pypqlx.py (hunk 2 above)
Comment (ebrasilio) [reply pypqlx.py]: ok, done!

File: pypqlx.py (hunk 3 above)
Comment (ebrasilio) [reply pypqlx.py]: Done!

File: pypqlx.py (hunk 4 above)
Comment (ebrasilio) [reply pypqlx.py]: Done!

**Pull request merged**